### PR TITLE
[Testing] Enable ci automated testing

### PR DESCRIPTION
**Pull request by bitparadigm**
Closes https://github.com/windranger-io/windranger-governance/issues/55.

### CODE REVIEW

**Comment by CjHare** [comment]
question: why are both the `push` and `pull_request` being run by this pull request.

Expected behaviour would be only the `pull_request` runs with a pull request, `push` being run on `main` when the PR is merged across.
e.g. Checks run in this PR https://github.com/windranger-io/windranger-treasury/pull/69

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (CjHare) [inline .github/dependabot.yaml:7]: suggestion: change the frequency for `github-actions` to `weekly`

Avoid Dependabot spammage.

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (CjHare) [inline .github/dependabot.yaml:11]: suggestion: change the frequency for `npm` to `monthly`

Avoid Dependabot spammage.

File: .github/workflows/master-push-pull.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Set up node
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Set ENV
+        run: mv .env.example .env
```
Comment (CjHare) [inline .github/workflows/master-push-pull.yaml:70]: suggestion: create a `.env` file specifically for the CI.

Repurposing the example for use by the CI, is unlikely to be implicitly known by people accessing the repo, so changing the example and having the CI break ....is unintuitive.

They may be a convention about having a `.env` file per an environment, but I'm not sure 🤷

**Review by CjHare — COMMENTED** [review]
It would have been good to have the Solidity static analyser working too.

Could you get that running locally? (i.e. it's a problem with GitHub action)

**Comment by bitparadigm** [comment]
I was able to run Slither locally. It indeed shows some 2 red suggestions and 3 yellow suggestions, which needs to be updated, but it's a whole separate PR and more than 10 green suggestions.

**Comment by bitparadigm** [comment]
Interesting that it shows red error in OZ code ```
TimelockController._call(bytes32,uint256,address,uint256,bytes) (node_modules/@openzeppelin/contracts/governance/TimelockController.sol#324-335) sends eth to arbitrary user
	Dangerous calls:
	- (success) = target.call{value: value}(data) (node_modules/@openzeppelin/contracts/governance/TimelockController.sol#331)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#functions-that-send-ether-to-arbitrary-destinations
``` so I might not be able to do anything about it.

**Comment by CjHare** [comment]
>whole separate PR 

Sounds good 👍 


>error in OZ code

When running `Slither` it'll be worthwhile excluding certain paths (such as OZ) with the `--filter-paths` option
e.g.
```
slither . --filter-paths "BitDAO.sol|node_modules"
```

**Review by CjHare — APPROVED** [review]

**Pull request merged**